Introduce test artifact for test utilities (#90)
ioki-mobility/kmp-passenger-api: README.md

However, there are more functions on `Result` like `map`, `successOrHandle`, `mapFailure`, etc.
Checkout the [`ioki-mobility/Result`](https://github.com/ioki-mobility/Result) documentation to find out more.

## Test artifact

> This project includes provides a test artifact that can be used as a test dependency in your project.

Reviewer: ```suggestion
This project provides a test artifact that can be used as a test dependency in your project.
```

> The test artifacts provides default implementation for the `Services` to create easy fakes for your tests.

Reviewer: ```suggestion
The test artifacts provide default implementations for the `Services` to easily create fakes for your tests. You can simply override the functions relevant for you and ignore the rest.
```

```kotlin
val fakeUserService = object : UserServiceFake() {
    // Only override what you need
    override suspend fun deleteUser(): ApiResult<Unit> = Result.Success(SuccessData(Unit))
}
```

Additionally, it provides helper functions to create APIObjects easily.
```kotlin
val rideResponse = createApiRideResponse()
```

> Those play well together for your tests:

Reviewer: ```suggestion
Those match well with your tests:
```

```kotlin
val fakeCurrentRideService = object : FakeCurrentRideService() {
    override suspend fun getCurrentRide(): ApiResult<ApiRideResponse> = Result.Success(SuccessData(
        createApiRideResponse(
            version = 1,
            pickup = createApiLocation(
                latitude = 50.1174225,
                longitude = 8.668939,
            ),
            dropoff = createApiLocation(
                latitude = 50.109694,
                longitude = 8.6666095,
            ),
        )
    ))
}

ClassUnderTest(fakeCurrentRideService)
```

To include it in your project, add the following dependency to your `build.gradle.kts` file:
```kotlin
val commonTest by getting {
    dependencies {

>         implementation("com.ioki:passenger-api-test:$currentVersion")

Reviewer: Maybe? 🤔 
```suggestion
        testImplementation("com.ioki:passenger-api-test:$currentVersion")
```

Author: The dependency declaration is different on KMP project.
Instead of having different `XXXimplementation()` declaration, they scope the declaration on the "`sourceSet`".
In this case, we're getting the `commonTest` sourceSet and declare the dependencies **just for that**.

Since `commonTest` *is the test sourceSet*, this is correct how it is declared 🙃
